scrollbar: Remove vertical scrollbar for alert box

m-e-l-u-h-a-n opened the pull request:
There was an unnecessary vertical scrollbar for the alert box during
the server reload. This was automatically inserted by browser due to
overflow-y: auto css rule on it. 

Fixes: #16911
**Testing plan:** <!-- How have you tested? -->
Testing has been done manually on `google chrome version: 87.0.4280.88` and `mozilla firefox version:83.0` by modifying the `event_register` so that alert box persists and then varying the screen width.

**GIFs or screenshots:** <!-- If a UI change.  See:
  https://zulip.readthedocs.io/en/latest/tutorials/screenshot-and-gif-software.html
  -->
Following were the changes after fixing:

**Chrome**
![chrome](https://user-images.githubusercontent.com/63504956/102475613-13f8dc00-4080-11eb-806e-ae164c211f11.png)

**fire fox**
![mozilla](https://user-images.githubusercontent.com/63504956/102475655-2541e880-4080-11eb-85a9-cabeb1ef46cc.png)

Actually this was only a problem in google chrome browser and was already fine in mozilla firefox

andersk commented:
We can’t just delete this scrollbar. If there are many simultaneous alerts or a long stacktrace in the dev environment, the scrollbar is needed to be able to scroll through them. Instead, we need to figure out and fix the reason it shows up when it’s not needed.

m-e-l-u-h-a-n commented:
@andersk As you suggested for display of scroll bar only if content exceeds the viewport, for that I have added a function in `index.html` to check continuously if bottom of `.alert-box` exceeds the screen and then accordingly add overflow property to achieve desirable display of scrollbar.
<!-- The style rules in css were already correct which can be confirmed by the fact that problem was only in some browsers(e.g. google chrome). -->
Changes:
**Normal**
![localhost_9991_](https://user-images.githubusercontent.com/63504956/102926183-96d3c980-44ba-11eb-8676-03492a48779c.png)

**With extra content**
![localhost_9991_ (1)](https://user-images.githubusercontent.com/63504956/102926237-ab17c680-44ba-11eb-86e5-d0f64b76e13f.png)

**With so many alerts**
![localhost_9991_ (2)](https://user-images.githubusercontent.com/63504956/102926303-bec32d00-44ba-11eb-9688-4f0a63a4a1a5.png)

Please review the changes

m-e-l-u-h-a-n commented:
Also some of the frontend tests are failing as I have added the function in the template itself and that violates the lint. I searched a lot in the documentation about appropriate  way to fix it(may be by placing function in some other file) but could not find one can you please guide me over solving it.

andersk reviewed (changes requested):
No. Please never write a continuous polling loop like this, in any application—it drains the user’s battery, which is a much more severe bug than the one you’re trying to fix.

This issue should be solved in pure CSS.

m-e-l-u-h-a-n commented:
Thanks for valuable suggestion, I will try something in pure CSS.

m-e-l-u-h-a-n commented:
@andersk I have added some extra browser specific css rules which hide the vertical scroll bar but allow user to scroll easily through the overflowed content in .alert-box. I have added a `.gif` for demo.
Since `max-height: 100%` is already applied in the `.alert-box` (which is the best way to display it as it's height varies with content), I think it could be decent solution to set display of scroll bar to none but allow scrolling.
Please have a look.
![ezgif com-video-to-gif](https://user-images.githubusercontent.com/63504956/103174036-dae11880-4884-11eb-8900-0d83ef55d66d.gif)

m-e-l-u-h-a-n commented:
I think pr [#16968](https://github.com/zulip/zulip/pull/16968) from @garg3133  fixes the issue in better way. You may have look into that.
